gsl/span

```diff
@@ -163,6 +163,8 @@ namespace details
         {
         }
 
+        constexpr span_iterator<Span, IsConst>& operator=(const span_iterator<Span, IsConst>&) = default;
```

Comment:
I think that is intentional to allow casting iterators pointing to nonconst values to an iterator pointing to const values.  I did fix this just now to be explicitly noexcept.